gateway/upstreamer/push/upstreamer.go

```diff
@@ -226,3 +240,32 @@ func (c *Upstreamer) listenService(ctx context.Context, ready chan struct{}) {
 		}
 	}
 }
+
+// Collect implement the FeedBackLoop interface to add new
+// samples into the feedbackloop
+func (c *Upstreamer) Collect(address string, responseTime time.Duration) {
```

Comment:
Why is the upsteamer implementing this? You only use the feedback loop in here. This should move there to avoid unnecessary dependencies